projects/samples/robotbenchmark/humanoid_sprint/controllers/race_stopwatch/race_stopwatch.c

```diff
@@ -84,14 +84,20 @@ int main(int argc, char *argv[]) {
   WbFieldRef translation = wb_supervisor_node_get_field(nao, "translation");
   int run = 1;
   double record = 0;
-  double penalty = 0;
+  bool disqualified = false;
+  const double time_limit = 5999.99;  // 99 min 59 s 99'
   do {
+    double t = wb_robot_get_time();
     const double *v = wb_supervisor_field_get_sf_vec3f(translation);
-    if (v[1] < 0.2) {  // the robot has fallen down
-      printf("The robot is down, a penalty of 30 seconds is added.\n");
-      penalty += 30;
+    if (!disqualified && v[1] < 0.2) {  // the robot has fallen down
+      printf("The robot is down, the robot is disqualified.\n");
+      disqualified = true;
+      stopwatch_set_time(digit, time_limit);
+      wb_robot_wwi_send_text("stop");
+      run = 0;
     }
-    double t = wb_robot_get_time() + penalty;
+    if (disqualified || t > (time_limit - time_step / 1000.0))
```

Comment:
`time_limit - time_step / 1000.0` is a constant value, isn't it?
it would probably be better to not recompute it at each time step